music.php

```diff
@@ -13,23 +13,28 @@
 		$dl = urldecode(trim($_GET['dl'], '/'));
 		if (is_dir($dl)) {
 			if (!chdir($dl)) die('Could not open folder: '. $dl);
-			$cmd = substr(php_uname(), 0, 7) == "Windows" ?
-				'7za a -tzip -mx1 -so . 2>&1' :	// -an -bd
-				'zip -0 -r - . 2>&1';	// Or 'zip -0 -j - * 2>&1' for non-recursive
-			$fp = popen($cmd, 'rb');
-			if (!$fp) die('Error creating zip');
-			header('Content-type: application/zip');
-			header('Content-disposition: attachment; filename="'. basename($dl) .'.zip"');
-			fpassthru($fp);
-			pclose($fp);
-			exit;
+			return_zip($dl, ['.'], false);
 		} elseif (file_exists($dl)) {
 			header('Content-type: '. mime_content_type($dl));
 			header('Content-Disposition: attachment; filename="'. basename($dl) .'"');
 			readfile($dl);
 			exit;
 		} else die('File not found: '. $dl);
-	} elseif (isset($_GET['pl'])) {
+	} 
+	elseif (isset($_GET['dlpl'])) {
+		$plname = urldecode($_GET['dlpl']);
+		$plfile = $cfg['playlistdir']. '/'. $plname. '.mfp.json';
+		if (!file_exists($plfile)) die('Playlist not found: '. $plfile);
+		$pl = (array) json_decode(stripcslashes(trim(file_get_contents($plfile), '"')));
+		$files = array();
+		foreach($pl['playlist'] as $song) { 
+			$path = $cfg['root']. '/'. ((array)$song)['path'];
+			if (!file_exists($path)) die('Playlist file not found: '. $path);

```

Comment:
I would describe $plfile as `Playlist file`, while in this loop you're talking about a song that can't be found.